### PR TITLE
Epic6.5 jordanh sprint1

### LINKED ISSUE #1325: appTheme.json build race condition
## Issue - Bug

So there is [still](https://circleci.com/gh/ParabolInc/action/3340) a rather annoying race condition in our build process that intermittently causes builds to fail.

The root cause of this is, if unlucky, appTheme.json can be deleted just as one of `build:server` or `build:client` is opening it.

To fix this, I believe we should:

- [ ] create yet another npm run command, this time calling it `build:theme`
- [ ] strip `WebpackShellPlugin` from our webpack configuration
- [ ] add `build:theme` to the `build` command

### PR DESCRIPTION
**Pull request by jordanh**
Fixes #1325 

It:

- [ ] Builds `appTheme.json` from `npm run build:theme`
- [ ] Builds all build dependencies concurrently when you run `npm run build`

### CODE REVIEW

**Comment by jordanh** [comment]
@ackernaut @mattkrick small change to the build process to eliminate the build race condition. Your 👀  requested...

File: package.json [hunk 1]
```diff
@@ -17,18 +17,20 @@
...
     "dev": "if [ ! -f \"dll/vendors.json\" ] ; then npm run build:dll ; fi ; NODE_ENV=development node --trace-warnings ./src/server/server.babel.js",
-    "build": "rimraf build && npm run build:relay && concurrently \"npm run build:client\" \"npm run build:server\"",
+    "build": "rimraf build && npm run build:deps && concurrently \"npm run build:client\" \"npm run build:server\"",
     "build:deploy": "rimraf build && npm run build:server && npm run build:client-deploy",
     "build:min": "rimraf build && concurrently \"npm run build:client-min\" \"npm run build:server\"",
     "bs": "npm run build && npm start",
     "debug:stripe": "node ./src/server/billing/debug.babel.js",
     "quickstart": "npm run db:migrate && npm run build && npm run start",
     "build:client": "NODE_ENV=production webpack --config ./webpack/prod.babel.js",
+    "build:deps": "concurrently \"npm run build:theme\" \"npm run build:dll\" \"npm run build:relay\"",
```
Comment (mattkrick) [inline package.json]: dll is only used for local development. why would we build it in prod?

File: webpack/utils/buildThemeJSON.babel.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+require('babel-register'); // eslint-disable-line import/no-extraneous-dependencies
+module.exports = require('./buildThemeJSON');
```
Comment (mattkrick) [inline webpack/utils/buildThemeJSON.babel.js]: we should remove the lockfile dance if we are only building it once

File: package.json (hunk 1 above)
Comment (jordanh) [reply package.json]: Was mostly a DX change. I can strike it.

File: webpack/utils/buildThemeJSON.babel.js (hunk 2 above)
Comment (jordanh) [reply webpack/utils/buildThemeJSON.babel.js]: I can do that.

**Comment by jordanh** [comment]
Made some changes @mattkrick – looking ok now?

**Pull request merged**